### DISCUSSION TITLE
What can trigger the "appIntegrity" callback

### DISCUSSION THREAD

**Discussion by LydGol90** [talsec/Free-RASP-ReactNative#43] [Q&A]
Hi there, we recently added the freeRASP react native SDK to our product and set up the useFreeRasp hook, adding some tracking for each callback just so that we could get some understanding of how many users might be affected by certain security fixes. We are seeing approximately 30% of our live android users are triggering in the "appIntegrity" callback. From what I could tell from the docs this is something to do with app signing? Would it be expected that this many users would have this vulnerability? Or could there be some other reasons here. Many thanks, Lydia

**Comment by tompsota** [comment]
Hi @LydGol90,

could you send us your watcherMail and package name at [support@talsec.app](mailto:support@talsec.app) together with reference to this discussion? We will look at the data.

Thanks,
Tomas, Talsec developer

**Comment by LydGol90** [comment]
@tompsota Thank you, I have done this!

**Comment by LydGol90** [comment]
@tompsota I've just reviewed the docs again and I didn't encode the certificate hash as directed, so I assume this is what is causing the issue!
